slack/slack.go

```diff
@@ -14,17 +15,26 @@ const DEFAULT_USERNAME = "ankh"
 
 // Send out a release message based on the chart, version and environment
 // supplied by the user
-func PingSlackChannel(ctx *ankh.ExecutionContext) error {
+func PingSlackChannel(ctx *ankh.ExecutionContext, ankhFile *ankh.AnkhFile) error {
 
 	// attempt the connection
 	api := slack.New(ctx.AnkhConfig.Slack.Token)
 
 	// get environment from env vs. context
-	deploymentEnvironment := util.GetEnvironmentOrContext(ctx.Environment, ctx.Context)
-	messageText, err := getMessageText(ctx, deploymentEnvironment)
-	if err != nil {
-		ctx.Logger.Infof("Unable to prompt for slack message. Using default value. Error: %v", err)
+	envOrContext := util.GetEnvironmentOrContext(ctx.Environment, ctx.Context)
+
+	var messages []string
+	for i := 0; i < len(ankhFile.Charts); i++ {
+		chart := &ankhFile.Charts[i]
+		message, err := getMessageText(ctx, chart, envOrContext)
+		if err != nil {
+			ctx.Logger.Infof("Unable to prompt for slack message. Using default value. Error: %v", err)
```

Comment:
let's return `fmt.Errorf(...`

Reply:
would this replace `ctx.Logger.Infof(...`?

Reply:
yeah, we'll return a formatted error and let the caller print it